libr/anal/var.c

```diff
@@ -346,38 +362,48 @@ struct VarUsedType {
 	st64 delta;
 };
 
-// TODO: this access-tracking should be per-function and also probably not in sdb
-R_API RAnalVar *r_anal_get_used_function_var(RAnal *anal, ut64 op_addr, R_NULLABLE RAnalFunction **fcn_out) {
-	char *inst_key = r_str_newf ("inst.0x%"PFMT64x".vars", op_addr);
-	const char *var_def = sdb_const_get (anal->sdb_fcns, inst_key, 0);
-	struct VarUsedType vut;
-	RAnalVar *res = NULL;
-	RAnalFunction *fcn = NULL;
-	if (sdb_fmt_tobin (var_def, SDB_VARUSED_FMT, &vut) == 4) {
-		fcn = r_anal_get_function_at (anal, vut.fcn_addr);
-		if (fcn) {
-			res = r_anal_function_get_var (fcn, vut.type[0], vut.delta);
-		}
-		sdb_fmt_free (&vut, SDB_VARUSED_FMT);
+R_API R_BORROW RPVector *r_anal_function_get_vars_used_at(RAnalFunction *fcn, ut64 op_addr) {
+	r_return_val_if_fail (fcn, NULL);
+	return ht_up_find (fcn->inst_vars, (st64)op_addr - (st64)fcn->addr, NULL);
```

Comment:
Ill probably put that substraction in a local const variable for readability. And im still unhappy with that signed cast but we can work on thst later in a separate pr if we see issues. Thanks for writing the tests